packages/client/ui/react-ui/src/components/auth/methods/email/EmailOTPInput.tsx

```diff
@@ -0,0 +1,114 @@
+import { InputOTP, InputOTPGroup, InputOTPSlot } from "@/components/common/InputOTP";
+import { useAuthSignIn } from "@/hooks/useAuthSignIn";
+import { EmailOtpIcon } from "@/icons/emailOTP";
+import { useState } from "react";
+import { useAuthForm } from "@/providers/auth/AuthFormProvider";
+import { AuthFormBackButton } from "../../AuthForm";
+
+export const EMAIL_VERIFICATION_TOKEN_LENGTH = 6;
+
+export function EmailOTPInput() {
+    const { appearance, baseUrl, apiKey, fetchAuthMaterial, setDialogOpen, setStep, otpEmailData, setOtpEmailData } =
```

Comment:
otp state should be inside email sign in as other methods don't use it

Reply:
I moved otp state and also email state out of that hook because i agree 👍